CIP-0006 | Adjust preamble and structure w.r.t CIP-0001 (#645)
cardano-foundation/CIPs: CIP-0006/README.md

---
CIP: 6
Title: Stake Pool Extended Metadata
Status: Active

> Category: Tools

Reviewer: ```suggestion
Category: Metadata
```
Happy for it to be either 🤓

Author: I'm glad you're happy with `Tools` then 🤓 since I already thought about which of these choices to make.  For other other CIPs tagged `Metadata` that metadata, even when off-chain, is associated with the chain somehow & in this case it isn't.

Also the target is a broad spectrum of operational procedures and tools as summarised best in this sentence (which itself could be a good description of the `Tools` category):
> Many additional attributes can be envisioned for future wallets, pool explorers, and information aggregators.

Reviewer: > For other CIPs tagged Metadata, their metadata - even when off-chain - is associated with the chain somehow, and in this case it isn't.

Makes sense to me! agreed.

Authors:
  - Markus Gufler <gufmar@gmail.com>
  - Mike Fullman <mike@fullman.net>
Implementors:
  - pooltool.io
  - cexplorer.io
Discussions:
  - https://github.com/cardano-foundation/CIPs/pull/15
Created: 2020-07-20
License: CC-BY-4.0
---

## Abstract

This CIP defines the concept of extended metadata for pools that is referenced from the pool registration data stored on chain.

## Motivation: why is this CIP necessary?

> As the ecosystem around Cardano stake pools proliferate so will the desire to slice, organize and search pool information dynamically.  Currently the metadata referenced on chain provides 512 bytes that can be allocated across the four information categories ([delegation-design-specification Section 4.2)](https://github.com/input-output-hk/cardano-ledger/releases/latest/download/shelley-delegation.pdf):

Reviewer: ```suggestion
As the ecosystem around Cardano stake pools proliferate so will the desire to slice, organize and search pool information dynamically.  Currently the metadata referenced on chain provides 512 bytes that can be allocated across the four information categories ([delegation-design-specification Section 4.2)](https://github.com/IntersectMBO/cardano-ledger/releases/latest/download/shelley-delegation.pdf):
```
do we want to do this?

Author: @Ryun1 @Crypto2099 @KtorZ two possible ways of answering your question (although definitely **yes** we want to correct the out of date links):

- We want to change these links to point to IntersectMBO because the IOHK links have become non-canonical and eventually the redirects will stop working (or get lost behind further redirects someday).
- We want to use the versions in our own repos since the academic releases of these technical papers are generally paywalled or at least require academic accounts to view them.

## Specification

> **Note** Updated: 2020-11-24 2nd key-pair for validation 2021-02-08 json schema

> ### On Chain referenced (main) metadata file

Reviewer: Are we going to enforce the any CIP which has on-chain data must provide a CDDL file (from #563) - retroactively?
Probably not practical.

Author: I think it would make sense to demand CDDL here if there had been a CIP for the original on-chain pool metadata specification, which there wasn't: as far as I know that was only ever documented in apocrypha like SPO setup procedures.

So providing CDDL for the extended structure, but not the original structure, doesn't generally solve that problem.

The alternative would be to broaden the scope of this CIP so it includes _both_ the original metadata spec _and_ the extended spec.  That would avoid having to write an "informational" CIP over 4 years after this method was put into practice: looking a bit silly perhaps.  I would still be in favour of it though: but only in a separate PR because it's beyond the scope of the CIP1 remediation effort.

This re-registration of the main metadata file with the `extData.vkey` and the two URLs is only necessary once. Afterwards, the operator can update his extended metadata at any time, compute the new signature with the `cardano-cli stake-pool rawdata-sign` command, and publish both files at the existing `extDataUrl` and `extSigUrl`.

## Rationale: how does this CIP achieve its goals?

> In the following we describe a first minimal version of the extended JSON file format

Reviewer: ```suggestion
In the following we describe a first minimal version of the extended JSON file format.
```
grammar :)

Author: @Ryun1 the period (rather than a colon) isn't a grammar mistake.  Mainly, it's a complete and grammatically correct sentence.  Also, the colon would normally be used if the "following" were following immediately, which it doesn't... instead it's distributed through this whole section & that's why I used the `.` rather than the `:`. 😛

Reviewer: youre right!!
